go/pkg/digest/digest.go

```diff
@@ -166,6 +166,34 @@ func NewFromReaderWithBuffer(r io.Reader, buf []byte) (Digest, error) {
 	}, nil
 }
 
+// CheckCapabilities returns an error if the digest function is not supported
+// by the server.
+func CheckCapabilities(caps *repb.ServerCapabilities) error {
+	fn := GetDigestFunction()
+
+	if caps.ExecutionCapabilities != nil {
+		if serverFn := caps.ExecutionCapabilities.DigestFunction; serverFn != fn {
+			return fmt.Errorf("server requires %v, client uses %v", serverFn, fn)
+		}
+	}
+
+	if caps.CacheCapabilities != nil {
+		cc := caps.CacheCapabilities
+		found := false
+		for _, serverFn := range cc.DigestFunction {
+			if serverFn == fn {
+				found = true
```

Comment:
it is fine to return from here?

Reply:
i think this makes the code a bit less robust. If another programmer doesn't read the code carefully and adds another check in the end of this func, it would be ignored.

Reply:
I see.